### ISSUE TITLE
Dropdown Component - Keyboard Navigation and accessibility

### ISSUE THREAD

**Issue by tigar** [ant-design/ant-design#24173]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
The current implementation of Dropdown is not accessible for keyboard users.

Features that would improve keyboard navigation:
escape key to close - when the menu is open pressing ESC to close the current dropdown
arrow key nav - use up/down arrow keys to traverse options in the dropdown

### What does the proposed API look like?
Adding a keydown handler to Dropdown similar to Cascader or Select:
```
  handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
    // SPACE => https://github.com/ant-design/ant-design/issues/16871
    if (e.keyCode === KeyCode.BACKSPACE || e.keyCode === KeyCode.SPACE) {
      e.stopPropagation();
    }
  };
```

**Comment by afc163** [comment]
same as #14356

**Comment by afc163** [comment]
- [ ] Cascader https://github.com/ant-design/ant-design/issues/16767 https://github.com/ant-design/ant-design/issues/17422

**Comment by simonauner** [comment]
I can see that keyboard navigation and selecting items with Enter now works on the documentation site https://ant.design/components/select/ . Even the search (Jack, Lucy, Tom) works as intended. However, when running 4.8.4 locally it's not working. Are there additional libs or a newer internal version running on the documentation site?

**Comment by afc163** [comment]
Fixed in https://github.com/ant-design/ant-design/pull/30382

**Comment by codewitch** [comment]
this isnt actually fixed by that PR. keyboard navigation works for regular menus but not for dropdowns
